lib/FusionInventory/Agent/Task/Inventory/Win32/License.pm

```diff
@@ -100,19 +91,28 @@ sub _getWmiSoftwareLicensingProducts {
             FULLNAME  => $object->{'Description'},
             NAME      => $object->{'Name'}
         };
-
-        if ($seenKey) {
-            # Update FULLNAME if seen ProductName in registry
-            $license->{FULLNAME} = encodeFromRegistry($seenKey->{'/ProductName'})
-                if $seenKey->{'/ProductName'};
-            $license->{TRIAL} = 1
-                if ($seenKey->{'/ProductNameBrand'} && $seenKey->{'/ProductNameBrand'} =~ /trial/i);
+   
+        if ($object->{'ID'} && $seenProducts->{lc($object->{'ID'})}) {
+            my $seenKey = $seenProducts->{lc($object->{'ID'})};
+            if ($seenKey->{'/ProductCode'}) {
+                my $ProductCodeUuid =  lc( $seenKey->{'/ProductCode'} =~ /([-\w]+)/ && $1 );
+                # Check Office registry is true
+                if ($seenProducts->{$ProductCodeUuid} && $seenProducts->{$ProductCodeUuid}->{'/DigitalProductID'}) {
+                    # Find Office registry licence
+                    my $indexSeenLicence = first {@{$licences}[$_]->{PRODUCTID} eq $seenProducts->{$ProductCodeUuid}->{'/ProductID'}} 0..$#$licences;
+                    # Registry Licence is good, it supply more information, so we skip WMI Office Licence
+                    next if defined $indexSeenLicence && @{$licences}[$indexSeenLicence]->{KEY} =~ m/$partialProductKey$/;
+                    # Registry is wrong, Wrong licence is deleted
+                    splice @{$licences}, $indexSeenLicence, 1 if defined $indexSeenLicence;
```

Comment:
I definitively not like working on indexes here, this is difficult to read and make code maintainability more complex.
Can you try better this simplified code ?
```
my $seenLicence = first { $_->{PRODUCTID} eq $seenProducts->{$ProductCodeUuid}->{'/ProductID'}} @{$licenses};
next if $seenLicence && $seenLicence->{KEY} =~ m/$partialProductKey$/;
@{$licenses} = grep { $_ != $seenLicence } @{$licenses} if $seenLicence;
```
Reading all the code again, I'm even thinking we may should better store all the licenses in $seenProducts, and work on that list (modifying _scanOfficeLicences() to store the final license object in place of the registry entries). Then this may simplify your tests. Finally we don't have to return @licenses arrays, and L.48 we would want to iterate other `values(%{$seenProducts})`. By the way, we may limit this PR to your fix, and I'll try to refactor thi for a later version.